Support for Kent Scientific KSC2024 Graphics Demo board

smammy opened the pull request.

smammy committed:
board: kent: gfxboard from sama5d27_som1_ek (1/3)

This is patch one of three creating the kent_gfxboard u-boot port.

This patch shows files that were copied from the Atmel sama5d27_som1_ek
port to create the basis of the kent_gfxdemo port. No changes have been
made in these files yet.

My hope is that by representing these copies as a separate commit, it
will make it easier to grok the differences between sama5d27_som1_ek and
kent_gfxdemo.

The copied files are as follows:

arch/arm/dts/
    at91-sama5d27_som1_ek.dts -> kent_gfxdemo.dts
board/atmel/
    sama5d27_som1_ek/Kconfig            -> kent/gfxdemo/Kconfig
    sama5d27_som1_ek/MAINTAINERS        -> kent/gfxdemo/MAINTAINERS
    sama5d27_som1_ek/Makefile           -> kent/gfxdemo/Makefile
    sama5d27_som1_ek/sama5d27_som1_ek.c -> kent/gfxdemo/kent_gfxdemo.c
configs/
    sama5d27_som1_ek_mmc_defconfig -> kent_gfxdemo_defconfig
include/configs/
    sama5d27_som1_ek.h -> kent_gfxdemo.h

smammy committed:
board: kent: gfxboard changes (2/3)

This is patch two of three creating the kent_gfxboard u-boot port.

This patch shows the modifications that were made to the Atmel
sama5d27_som1_ek port to represent the hardware present on the Kent
board. These files were copied from the corresponding sama5d27_som1_ek
files in the previous patch.

The functional changes are:

- dts:
    - remove PDA LCD connections and 1-wire ID EEPROM
    - remove USB power control
    - Use 4-wire bus width instead of 8 for sdmmc0
    - remove sdmmc1
- board C file:
    - remove USB power control
    - change pin assignments for LEDs
- defconfig:
    - SUPPORT_RAW_INITRD=y to load bare cpio archive generated by yocto

smammy committed:
board: kent: gfxboard wiring (3/3)

This is patch three of three creating the kent_gfxboard u-boot port.

This patch includes the Kconfig and Makefile "wiring" necessary to
enable the port. The changes are:

- dts: add dtb-$(CONFIG_TARGET_KENT_GFXDEMO) variable.
- mach-at91: add TARGET_KENT_GFXDEMO config item.

malsyned commented on arch/arm/dts/kent_gfxdemo.dts:
```diff
@@ -0,0 +1,121 @@
+	compatible = "atmel,kent-gfxdemo", "atmel,sama5d2", "atmel,sama5";
+
+	chosen {
+		bootph-all;
```
I don't remember needing `bootph-all` being a part of my life last time I did this. That doesn't mean you don't need it -- is it something you got from copying the ek1 dts?

smammy replied:
Yes, it is, and I haven't tested without it. Shall I do it now, or shall we put it in the backlog?

malsyned replied:
Nah, I'm not asking for a change here, I'm just curious if you knew anything about it.

malsyned commented on board/kent/gfxdemo/MAINTAINERS:
```diff
@@ -0,0 +1,6 @@
+KENT GFXDEMO BOARD
+M:	Sam Birch <sbirch@innovative-electronics.com>
```
Do you want to use your profirmserv.com address for this?

smammy replied:
I'm actually kinda unclear on my "identity" for the purposes of this file and also for the copyright statements and email addresses there. I've been putting IELLC and using my IELLC address, but since (I assume) this is work for hire, it should probably be ©Kent and, sure, I should like to use sam@profirmserv.com address.

smammy replied:
Also I should mention that I'm pretty skeptical that devicetree files are eligible for copyright protection, but here we are.

smammy replied:
I have normalized copyright and license statements.

malsyned replied:
OK, be all that as it may, I don't think this particular line of code is a copyright statement. It's a maintainer pointer.

smammy committed:
board: kent: correct copyright/license info

Code will be owned by Kent and I am using my sam@profirmserv.com email.
I also added/fixed the SPDX License IDs.

smammy committed:
board: kent: normalize whitespace
